iroh-bytes/src/store/traits.rs

```diff
@@ -103,23 +110,178 @@ pub trait Map: Clone + Send + Sync + 'static {
 
 /// A partial entry
 pub trait PartialMapEntry<D: PartialMap>: MapEntry<D> {
-    /// A future that resolves to an writeable outboard
-    fn outboard_mut(&self) -> BoxFuture<'_, io::Result<D::OutboardMut>>;
-    /// A future that resolves to a writer that can be used to write the data
-    fn data_writer(&self) -> BoxFuture<'_, io::Result<D::DataWriter>>;
+    /// Get a batch writer
+    fn batch_writer(&self) -> BoxFuture<'_, io::Result<D::BatchWriter>>;
+}
+
+/// An async batch interface for writing bao content items to a pair of data and
+/// outboard.
+///
+/// Details like the chunk group size and the actual storage location are left
+/// to the implementation.
+pub trait BaoBatchWriter {
+    /// Write a batch of bao content items to the underlying storage.
+    ///
+    /// The batch is guaranteed to be sorted as data is received from the network.
+    /// So leafs will be sorted by offset, and parents will be sorted by pre order
+    /// traversal offset. There is no guarantee that they will be consecutive
+    /// though.
+    ///
+    /// The size is the size the remote side told us. It is not guaranteed
+    /// to be correct, but it is guaranteed to be consistent with all data in the
+    /// batch.
+    ///
+    /// Batches should not become too large. Typically, a batch is just a few
+    /// parent nodes and a leaf.
+    ///
+    /// Batch is a vec so it can be moved into a task, which is unfortunately
+    /// necessary in typical io code.
+    ///
+    /// TODO: return impl Future<...> once we got 1.75
+    fn write_batch(
+        &mut self,
+        size: u64,
+        batch: Vec<BaoContentItem>,
+    ) -> LocalBoxFuture<'_, io::Result<()>>;
+
+    /// Sync the writer
```

Comment:
maybe specify to what this syncs, like that it persists or so, and not synced to another node ;)